src/sync/remote_mongo_collection.hpp

```diff
@@ -0,0 +1,184 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_CORE_REMOTE_MONGO_COLLECTION_HPP
+#define REALM_CORE_REMOTE_MONGO_COLLECTION_HPP
+
+#include <string>
+#include <vector>
+#include <realm/util/optional.hpp>
+#include <json.hpp>
+#include "remote_mongo_read_operation.hpp"
+#include "remote_mongo_result_types.hpp"
+
+namespace realm {
+namespace app {
+
+template<class T>
+class RemoteMongoCollection {
+    
+public:
+    
+    using CollectionType = T;
+    using Document = nlohmann::json;
```

Comment:
[minor] Instead of this, make it a `std::string`, and change the parameter names to be suffixed with `_json`

Reply:
Addressed